File: api_multicomercio/app/Models/Direccion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Direccion extends Model
{
    use HasFactory;
    protected $table = 'direcciones';

    protected $fillable = [
        'user_id',
        'nombre',
        'direccion',
        'referencia',
        'latitud',
        'longitud',
        'predeterminada',
    ];


    //cast de los campos para que se conviertan automáticamente a boolean y float,
    //al leer los datos de la bd y los serealiza a json se convierten a los tipos correctos
    protected $casts = [
        'predeterminada' => 'boolean',
        'latitud'        => 'float',
        'longitud'       => 'float',
    ];

    // Usuario cliente al que le pertenece esta dirección
    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: app_pedidos_multicomercio/app/src/main/java/com/example/app_pedidos_multicomercio/Models/Direccion.java
package com.example.app_pedidos_multicomercio.Models;

public class Direccion {
    private int id;
    private String nombre;
    private String direccion;
    private String referencia;
    private Double latitud;
    private Double longitud;
    private boolean predeterminada;

    public int getId() { return id; }
    public String getNombre() { return nombre; }
    public String getDireccion() { return direccion; }
    public String getReferencia() { return referencia; }
    public Double getLatitud() { return latitud; }
    public Double getLongitud() { return longitud; }
    public boolean isPredeterminada() { return predeterminada; }

    @Override
    public String toString() {
        return nombre + " — " + direccion;
    }
}
